siobrultech_protocols/gem/protocol.py

```diff
@@ -47,17 +47,18 @@ def __init__(
         self._transport: Optional[asyncio.BaseTransport] = None
 
     def connection_made(self, transport: asyncio.BaseTransport) -> None:
+        LOG.info("Connection opened")
         self._transport = transport
 
-    def connection_lost(self, exc: Optional[Any]) -> None:
+    def connection_lost(self, exc: Optional[BaseException]) -> None:
         if exc is not None:
-            LOG.warning("Connection lost: {}".format(exc))
+            LOG.warning("Connection lost due to exception", exc)
```

Comment:
I think that, as this is written, the `exc` parameter is ignored. You either want to keep a `{}` somewhere in the format string, or use a keyword argument `exc_info=exc`. (https://docs.python.org/3/library/logging.html#logging.debug)